ql/src/java/org/apache/hadoop/hive/ql/exec/tez/WorkloadManager.java

```diff
@@ -486,15 +486,16 @@ private void scheduleWork(WmThreadSyncWork context) {
             LOG.info("Invoking KillQuery for " + queryId + ": " + reason);
             try {
               UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
-              SessionState ss = new SessionState(new HiveConf(), ugi.getShortUserName());
-              ss.setIsHiveServerQuery(true);
-              SessionState.start(ss);
-              kq.killQuery(queryId, reason, toKill.getConf());
-              addKillQueryResult(toKill, true);
-              killCtx.killSessionFuture.set(true);
-              wmEvent.endEvent(toKill);
-              LOG.debug("Killed " + queryId);
-              return;
+                try (SessionState ss = new SessionState(new HiveConf(), ugi.getShortUserName())) {
```

Comment:
looks legit to me. @abstractdog could you please confirm that's ok to close session object here?

Reply:
here, as I referred to in the case of the MV registry, what about trying a SessionState field in the workload manager, so instead of:
```
              UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
              SessionState ss = new SessionState(new HiveConf(), ugi.getShortUserName());
```
we can initialize the SessionState when the WorkloadManager is created, because neither the object returned by new HiveConf(), nor the UserGroupInformation.getCurrentUser() is supposed to change over time in HS2, it's safe to reuse SessionState